File: PCube-API/domain/project.py
class Project:

    def __init__(self,):
        self.id = -1
        self.name = ""
        self.parent_id = -1
        

    def asDictionnary(self):
        return {
            "id": self.id,
            "name": self.name,            
            "parent_id": self.parent_id,          
        }

File: PCube/src/app/models/project.ts
export class ProjectItem {
    id : number;
    name : string;
    parent_id: number;
    child_project?: ProjectItem[];
    nomAffichage? : string;

    constructor(projectResponse?: any) {
        this.id = projectResponse && projectResponse.id || null;
        this.name = projectResponse && projectResponse.name || "";
        this.parent_id = projectResponse && projectResponse.parent_id || null;
        this.nomAffichage = projectResponse && projectResponse.nomAffichage || "";
        this.child_project = projectResponse && projectResponse.child_project || [];
    }
}